### PR TITLE
Added solution to HW-SPR06

### PR DESCRIPTION
**Pull request by MichaelShevchenko**

### CODE REVIEW

**Review by fyarkiy — APPROVED** [review]
Good job!

File: src/main/java/com/dev/cinema/controller/ShoppingCartController.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+    @PostMapping("/movie-sessions")
+    public void addMovieSession(@RequestParam Long movieSessionId,
+                                @RequestParam Long userId) {
+        shoppingCartService.addSession(movieSessionService.getById(movieSessionId),
+                userService.getById(userId));
+    }
+
+    @GetMapping("/by-user")
+    public ShoppingCartResponseDto getByUserId(@RequestParam(name = "id") Long userId) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/controller/ShoppingCartController.java]:
```suggestion
    public ShoppingCartResponseDto getByUserId(@RequestParam Long userId) {
```
let's leave name userId(or you can change it to user_id if you want). just "id" would means that you are getting shopping cart by its own ID

File: src/main/java/com/dev/cinema/controller/UserController.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    private final UserService userService;
+
+    @Autowired
+    public UserController(UserMapper userMapper, UserService userService) {
+        this.userMapper = userMapper;
+        this.userService = userService;
+    }
+
+    @GetMapping("/by-email")
+    public UserResponseDto get(@RequestParam String email) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/controller/UserController.java]:
```suggestion
    public UserResponseDto getByEmail(@RequestParam String email) {
```

File: src/main/java/com/dev/cinema/dto/OrderRequestDto.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
+package com.dev.cinema.dto;
+
+import java.util.List;
+
+public class OrderRequestDto {
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/dto/OrderRequestDto.java]: do you need this class?

File: src/main/java/com/dev/cinema/dto/TicketResponseDto.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
+package com.dev.cinema.dto;
+
+public class TicketResponseDto {
+    private Long id;
+    private String movieTitle;
+    private Long userId;
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/dto/TicketResponseDto.java]: you already jave information about user in OrderResponseDto and ShoppingCartResponseDto. No need to add it to all TicketResponseDto(list of which is the part of both of above dtos)

File: src/main/java/com/dev/cinema/controller/ShoppingCartController.java (hunk 1 above)
Comment (MichaelShevchenko) [reply src/main/java/com/dev/cinema/controller/ShoppingCartController.java]: done

File: src/main/java/com/dev/cinema/controller/UserController.java (hunk 2 above)
Comment (MichaelShevchenko) [reply src/main/java/com/dev/cinema/controller/UserController.java]: done

File: src/main/java/com/dev/cinema/dto/OrderRequestDto.java (hunk 3 above)
Comment (MichaelShevchenko) [reply src/main/java/com/dev/cinema/dto/OrderRequestDto.java]: done

File: src/main/java/com/dev/cinema/dto/TicketResponseDto.java (hunk 4 above)
Comment (MichaelShevchenko) [reply src/main/java/com/dev/cinema/dto/TicketResponseDto.java]: done

File: src/main/java/com/dev/cinema/dto/TicketResponseDto.java [hunk 5]
```diff
@@ -21,17 +20,8 @@ public void setMovieTitle(String movieTitle) {
...
-
-    public void setUserId(Long userId) {
-        this.userId = userId;
-    }
-
     @Override
     public String toString() {
-        return "TicketResponseDto{" + "movieTitle='" + movieTitle + '\''
-                + ", userId=" + userId + '}';
+        return "TicketResponseDto{" + "movieTitle='" + movieTitle + '\'' + '}';
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/dto/TicketResponseDto.java]: two redundant concationaions

**Review by dmytro-hryhoruk — APPROVED** [review]

File: src/main/java/com/dev/cinema/dto/TicketResponseDto.java (hunk 5 above)
Comment (MichaelShevchenko) [reply src/main/java/com/dev/cinema/dto/TicketResponseDto.java]: done

**Pull request merged**